contrib/babelfishpg_tsql/src/rolecmds.c

```diff
@@ -1858,6 +1933,240 @@ get_roleform_ext(char *login)
 	return tuple;
 }
 
+/*
+* Utility function to validate netbios name provided by user
+*/
+
+static void
+validateNetBIOS(char* netbios)
+{
+	int len = strlen(netbios);
+	int i = 0;
+
+	if (len > NETBIOS_NAME_MAX_LEN || len < NETBIOS_NAME_MIN_LEN)
+		ereport(ERROR,
+			(errcode(ERRCODE_INVALID_NAME),
+				errmsg("The NetBIOS name '%s' has invalid length. NetBIOS name length should be between %d and %d.",
+				netbios, NETBIOS_NAME_MIN_LEN, NETBIOS_NAME_MAX_LEN)));
+
+	if (netbios[0] == '.')
+		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+			errmsg("'%s' is not a valid NetBIOS name. It must not start with '.' .", netbios)));
```

Comment:
Why is this scenario separately tested from the conditions mentioned in the while loop next?